File: src/main/java/javaf/pws/secondPW/SecondTask.java
package javaf.pws.secondPW;

import java.util.NoSuchElementException;
import java.util.Scanner;

public class SecondTask {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        try{
            System.out.println("enter a number");
            String number = scanner.nextLine();
            if (number.isEmpty() || number.isBlank()) System.out.println("Error: String is empty or blank");
            else {
                int n = Integer.parseInt(number);
                int sum = 0;
                while (n > 0) {
                    if (forTask(n)) sum += (n % 10);
                    n /= 10;
                }
                System.out.println(sum);
            }
        } catch (NoSuchElementException ex){
            System.err.println("Error: String is null");
        } catch (NumberFormatException ex){
            System.err.println("Error: This isn't a number");
        }
    }
    private static boolean forTask(int n){ return ((n % 10) % 3) != 0;}
}

File: src/main/java/kotlinf/pws/secondPW/secondTask.kt
package kotlinf.pws.secondPW

//В программе, сделанной в задании No1, вынесите проверяемое условие в отдельную single-expression функцию.
fun secondTask(it: Int) = (it % 3) != 0
fun secondTaskV1(n: Int): Int {
    var number = n
    var value: Int
    var sum = 0
    while (number > 0) {
        value = number % 10
        if (secondTask(value)) sum += value
        number /= 10
    }
    return sum
}
tailrec fun secondTaskV2(n: Int, sum: Int = 0): Int {
    if (n == 0) return sum
    val value = n % 10
    val newSum = if (secondTask(value)) sum + value else sum
    return secondTaskV2(n / 10, newSum)
}
fun main() {
    println("enter a number")
    val n = readlnOrNull()?.toIntOrNull()
    if ((n == null) || (n <= 0)) println("Error")
    else {
        println(secondTaskV1(n))
        println(secondTaskV2(n))
    }
}